src/io/util.jl

```diff
@@ -232,6 +232,23 @@ function get_row_idxs(table, pair::Pair)
 end
 export get_row_idxs
 
+"""
+    row_comparison(row::DataFrameRow, pairs)
+
+Compares a single DataFrameRow to the pairs given and returns the true/false result of the comparison. 
+"""
+function row_comparison(row::DataFrameRow, pairs)
+    result = true # start as true and will turn false if any of the pairs isn't met
```

Comment:
Looks like this should work!  However, I think we can do a little better performance-wise by short-circuiting the evaluation.  I.e. instead of allocating `result` here, we can simply return false if we encounter a false comparison.  That way we're not evaluating more comparisons than is needed.